clients/drcachesim/tools/view.cpp

```diff
@@ -248,6 +251,9 @@ view_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
             // We delay printing until we know the tid.
             if (trace_version_ == -1) {
                 trace_version_ = static_cast<int>(memref.marker.marker_value);
+                if (filetype_ != -1 && !init_from_stream()) {
```

Comment:
The logic is a little confusing, especially with the name "init_from_stream".  If we have the version and the filetype, why would we need to get them from the stream? 

Also, it looks like this doesn't need to call anything: I don't see the version used in init_from_filetype().
I would just remove these added lines here altogether.

Reply:
Done. Simplified.